lib/components/user/monitored-trip/trip-status.js

```diff
@@ -0,0 +1,391 @@
+        // exist for itineraries that have differing values of the scheduled
+        // departure time and matching itinerary departure time.
+        if (journeyState.hasRealtimeData) {
+          // calculate the deviation from the scheduled departure time (positive
+          // value indicates delay)
+          const departureDeviationSeconds = (
+            matchingItinerary.startTime -
+            journeyState.scheduledDepartureTimeEpochMillis
+          ) / 1000
+          const deviationHumanDuration = formatDuration(
+            Math.abs(departureDeviationSeconds)
+          )
+          if (Math.abs(departureDeviationSeconds) < ON_TIME_THRESHOLD_SECONDS) {
+            // about on time
+            headerText = 'Trip is starting soon and is about on time.'
+            panelBsStyle = 'success'
+          } else if (departureDeviationSeconds > 0) {
+            // delayed
+            headerText = `Trip start time is delayed ${deviationHumanDuration}!`
+            panelBsStyle = 'warning'
+          } else {
+            // early
+            headerText = `Trip start time is happening ${deviationHumanDuration} earlier than expected!`
+            panelBsStyle = 'warning'
+          }
+        } else {
+          headerText = 'Trip is starting soon (no realtime updates available).'
+          panelBsStyle = undefined
+        }
+      }
+
+      body = `Trip is due to begin at ${tripStartMoment.format(timeFormat)}.`
+    } else if (journeyState.tripStatus === 'TRIP_ACTIVE') {
+      // analyze whether the journey state indicates that the matching itinerary
+      // has realtime data. It is assumed that realtime data will exist for
+      // itineraries that have differing values of the scheduled arrival time
+      // and matching itinerary arrival time.
+      if (journeyState.hasRealtimeData) {
+        // calculate the deviation from the scheduled arrival time (positive
+        // value indicates delay)
+        const arrivalDeviationSeconds = (
+          matchingItinerary.endTime -
+          journeyState.scheduledArrivalTimeEpochMillis
+        ) / 1000
+        const deviationHumanDuration = formatDuration(
+          Math.abs(arrivalDeviationSeconds)
+        )
+        if (Math.abs(arrivalDeviationSeconds) < ON_TIME_THRESHOLD_SECONDS) {
+          // about on time
+          headerText = 'Trip is in progress and is about on time.'
+          panelBsStyle = 'success'
+        } else if (arrivalDeviationSeconds > 0) {
+          // delayed
+          headerText = `Trip is in progress and is delayed ${deviationHumanDuration}!`
+          panelBsStyle = 'warning'
+        } else {
+          // early
+          headerText = `Trip is in progress and is arriving ${deviationHumanDuration} earlier than expected!`
+          panelBsStyle = 'warning'
+        }
+      } else {
+        headerText = 'Trip is in progress (no realtime updates available).'
+        panelBsStyle = undefined
```

Comment:
I'm going to hold off on any further updates since that might require custom styling. This code currently just sets a bootstrap panel theme and the rest gets taken care of.